File: C#/Stack/2390. Removing Stars From a String.cs
/* 2390. Removing Stars From a String
Difficulty: Easy
Submission Link: https://leetcode.com/problems/removing-stars-from-a-string/submissions/1375561311/ */

public class Solution {
    public string RemoveStars(string s) {
        char[] result = new char[s.Length];
        int top = 0;

        foreach (char c in s) {
            if (c != '*') {
                result[top++] = c;
            } else {
                top--;
            }
        }
        return new string(result, 0, top);
    }
}

File: C/Stack/2390. Removing Stars From a String.c
/* 2390. Removing Stars From a String
Difficulty: Easy
Submission Link: https://leetcode.com/problems/removing-stars-from-a-string/submissions/1375561467/ */

char* removeStars(char* s) {
    int top=0,i;
    for(i=0;s[i];i++)
        (s[i]!='*')?(s[top++]=s[i]):top--;
    s[top]=NULL;
    return s;
}